### ISSUE TITLE
Fails when trying to attach to a nonexistent IPv6 interface

### ISSUE THREAD

**Issue by utrack** [gabe565/CastSponsorSkip#54]
Heya! That's a neat service, thank you :)

I've hit a problem running it on my laptop (both 'natively' and via Docker). It spams that line to the logs:
```
2023/12/11 21:44:33 ERROR Failed to discover devices. error="unable to create new zeroconf resolver: listen udp6 [ff02::]:5353: socket: address family not supported by protocol"
```

I'm running Arch Linux rolling, and I don't have any v6 interfaces. Here's the `ip a`:
```
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
2: wlan0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether 04:7b:cb:b3:4a:e0 brd ff:ff:ff:ff:ff:ff
    inet 192.168.50.123/24 brd 192.168.50.255 scope global dynamic noprefixroute wlan0
       valid_lft 84715sec preferred_lft 84715sec
4: docker0: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc noqueue state DOWN group default
    link/ether 02:42:00:a2:22:a0 brd ff:ff:ff:ff:ff:ff
    inet 172.4.0.1/24 brd 172.4.0.255 scope global docker0
       valid_lft forever preferred_lft forever
```

Cheers! Let me know if you need me as a guinea pig to test things :)

**Comment by gabe565** [comment]
@utrack Sorry for the late response, I was busy with the holidays and missed this issue! Were you ever able to work around this?

If not, this is an interesting one. Admittedly, I don't use IPv6 either and I haven't seen this issue. Let me test with an Arch device to see if I have the same error.

I'd maybe suggest installing [go-chromecast](https://github.com/vishen/go-chromecast), which is the the library this repo uses to communicate with devices. Once installed, maybe just something like `go-chromecast ls`?

**Comment by utrack** [comment]
Yep, I guess it's an issue on their side :) thanks! I'll report that to upstream.

```
> go-chromecast ls
error: unable to discover chromecast devices: unable to create new zeroconf resolver: listen udp6 [ff02::]:5353: socket: address family not supported by protocol
```

**Comment by FlyveHest** [comment]
+1 for this, happening on my system as well, I also haven't got ipv6 enabled.

**Comment by gabe565** [comment]
@FlyveHest Does it help if you bind a specific interface with the `--network-interface` flag or the `CSS_NETWORK_INTERFACE` env?

**Comment by gabe565** [comment]
@utrack I saw that go-chromecast is working again for you...is CastSponsorSkip also working?

**Comment by FlyveHest** [comment]
> @FlyveHest Does it help if you bind a specific interface with the `--network-interface` flag or the `CSS_NETWORK_INTERFACE` env?

Thats what I tried first, did not make a difference, but when I bound it to an interface (I have 4) and specified my devices IP (ChromeCast is on a static lease for firewalling purposes), it skipped discovery and just connected.

**Comment by utrack** [comment]
@gabe565 Yes, it does - I didn't do any rebuilds, so it seems like this issue happens with some specific OS setup.

**Comment by atesin** [comment]
this is a giant problem to me and the reason why i DONT USE CastSponsorSkip ... i think there are nearly 1000 of these logs messages per second, that fill my harddisk in a day, blocking my system... additionally i found no way to suppress these messages to being written in systemd journal

check #119

**Comment by gabe565** [comment]
@atesin I'm not sure why this is happening for some people. Did you try restricting the network interface that is bound to? I also don't use IPv6, and I don't receive these logs, so maybe you have an interface defined with an IPv6 address which isn't being used or something?

**Comment by gabe565** [comment]
I just updated go-chromecast to the latest commit and it looks like this has been fixed! I'll release v0.8.1 shortly. If any of you get a chance to test, please let me know if this error is resolved for you.

**Comment by gabe565** [comment]
Has anybody gotten the chance to test this? The issue should be resolved, but I'd like to verify before closing this issue.

**Comment by FlyveHest** [comment]
I just tried removing the direct IP binding, and it connects to the Chromecast as expected, so looks like the fix works. :)
